labs/aws/go/lab-04/code/step8.go

```diff
@@ -0,0 +1,265 @@
+	"github.com/pulumi/pulumi-kubernetes/sdk/go/kubernetes/providers"
+	"github.com/pulumi/pulumi/sdk/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Read back the default VPC and public subnets, which we will use.
+		t := true
+		vpc, err := ec2.LookupVpc(ctx, &ec2.LookupVpcArgs{Default: &t})
+		if err != nil {
+			return err
+		}
+		subnet, err := ec2.GetSubnetIds(ctx, &ec2.GetSubnetIdsArgs{VpcId: vpc.Id})
+		if err != nil {
+			return err
+		}
+		eksRole, err := iam.NewRole(ctx, "eks-iam-eksRole", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
+		    "Version": "2008-10-17",
+		    "Statement": [{
+		        "Sid": "",
+		        "Effect": "Allow",
+		        "Principal": {
+		            "Service": "eks.amazonaws.com"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+		})
+		if err != nil {
+			return err
+		}
+		eksPolicies := []string{
+			"arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+			"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+		}
+		for i, eksPolicy := range eksPolicies {
+			_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("rpa-%d", i), &iam.RolePolicyAttachmentArgs{
+				PolicyArn: pulumi.String(eksPolicy),
+				Role:      eksRole.Name,
+			})
+			if err != nil {
+				return err
+			}
+		}
+		// Create the EC2 NodeGroup Role
+		nodeGroupRole, err := iam.NewRole(ctx, "nodegroup-iam-role", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
+		    "Version": "2012-10-17",
+		    "Statement": [{
+		        "Sid": "",
+		        "Effect": "Allow",
+		        "Principal": {
+		            "Service": "ec2.amazonaws.com"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+		})
+		if err != nil {
+			return err
+		}
+		nodeGroupPolicies := []string{
+			"arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+			"arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+			"arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+		}
+		for i, nodeGroupPolicy := range nodeGroupPolicies {
+			_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("ngpa-%d", i), &iam.RolePolicyAttachmentArgs{
+				Role:      nodeGroupRole.Name,
+				PolicyArn: pulumi.String(nodeGroupPolicy),
+			})
+			if err != nil {
+				return err
+			}
+		}
+		// Create a Security Group that we can use to actually connect to our cluster
+		clusterSg, err := ec2.NewSecurityGroup(ctx, "cluster-sg", &ec2.SecurityGroupArgs{
+			VpcId: pulumi.String(vpc.Id),
+			Egress: ec2.SecurityGroupEgressArray{
+				ec2.SecurityGroupEgressArgs{
+					Protocol:   pulumi.String("-1"),
+					FromPort:   pulumi.Int(0),
+					ToPort:     pulumi.Int(0),
+					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+				},
+			},
+			Ingress: ec2.SecurityGroupIngressArray{
+				ec2.SecurityGroupIngressArgs{
+					Protocol:   pulumi.String("tcp"),
+					FromPort:   pulumi.Int(80),
+					ToPort:     pulumi.Int(80),
+					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
```

Comment:
Don't believe this is needed for any of the steps

Reply:
This is to access the app as port 80 is exposed by the service